.github/workflows/linux-eic-shell.yml

```diff
@@ -48,17 +48,19 @@ jobs:
       uses: actions/cache@v3
       with:
         path: .ccache
-        key: ccache-${{ matrix.CC }}-${{ matrix.CMAKE_BUILD_TYPE }}-${{ github.ref_name }}-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+        key: ccache-${{ matrix.CC }}-${{ matrix.CMAKE_BUILD_TYPE }}-${{ github.event.pull_request.head.ref || github.ref_name }}-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
```

Comment:
I'm not able to understand this. Could you comment on ref/ref_name use here?

Reply:
The idea here was to use the branch name in the key, not whatever github decides to call its temporary merge branches. E.g. for this PR, the cache key should be `ci-ccache-max-dir`, not `774/merge` or `gh-readonly-queue/main/pr-774`.

If we (as I'm prone to do) want to branch off from another branch that is not `main`, then adding the base branch ref as a lookup key only works if that has been used as a key for writing.

In other words, currently we write caches with key `main` or `774/mere` (pull request) or `gh-readonly-queue/main/pr-774` (merge queue), that's what's in `github.ref_name`. If we branch off of `ci-ccache-max-dir`, then the new branch has no way of knowing it should look for a cache with key `774/merge`. This changes the keys to use `main` or `ci-ccache-max-dir`.

Reply:
See https://github.com/eic/EICrecon/actions/caches?query=ccache for names of cached objects:
- ccache-gcc-Debug-gh-readonly-queue/main/pr-1006-7b1e880bc061f94a676ab46159a8cc1d2f3c11a2-2023-09-23T01:08+00:00
- ccache-gcc-Release-995/merge-2023-09-23T00:58+00:00
- ccache-gcc-Debug-main-2023-09-22T23:33+00:00

It's only for `main` that we actually use the branch name.

Reply:
I understand better now. Plus, I now see that only `base.ref` and `head.ref` are available for the pull_request event: https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request. From what I understand they are not fully qualified ref like `github.ref`, but more like `github.ref_name`.

There are also `github.base_ref` and `github.head_ref`, which should be synonyms to `github.even.pull_request.*.ref`. We don't use `pull_request_target`, so we don't really prefer either.